core/ops.rs

```diff
@@ -28,17 +28,25 @@ use std::task::Poll;
 /// turn of the event loop, which is too late for certain ops.
 pub struct OpCall<T>(MaybeDone<Pin<Box<dyn Future<Output = T>>>>);
 
+pub enum EagerPollResult<T> {
+  Ready(T),
+  Pending(OpCall<T>),
+}
+
 impl<T> OpCall<T> {
   /// Wraps a future, and polls the inner future immediately.
   /// This should be the default choice for ops.
-  pub fn eager(fut: impl Future<Output = T> + 'static) -> Self {
+  pub fn eager(fut: impl Future<Output = T> + 'static) -> EagerPollResult<T> {
     let boxed = Box::pin(fut) as Pin<Box<dyn Future<Output = T>>>;
```

Comment:
Potential optimization: defer the `Box::pin` for futures that are `Unpin`. Avoids a heap allocation when the future completes immediately.

You can probably also specialize for futures that are already `Box<Pin<dyn Future>>`, to avoid boxing them twice.

(Only worthwhile if such futures actually exist, of course. No point if everything is `!Unpin`.)